system-tests/tia-maven/src/test/java/com/teamscale/tia/TiaMavenMultipleJobsTest.java

```diff
@@ -0,0 +1,43 @@
+package com.teamscale.tia;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+import org.conqat.lib.commons.filesystem.FileSystemUtils;
+import org.conqat.lib.commons.io.ProcessUtils;
+import org.conqat.lib.commons.system.SystemUtils;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test class to check if multiple maven plugins can be started with dynamic
+ * port allocation.
+ */
+public class TiaMavenMultipleJobsTest {
+
+	/**
+	 * Starts multiple maven processes and checks that the ports are dynamically set
+	 * and the servers are correctly started.
+	 */
+	@Test
+	public void testMavenTia() throws Exception {
+		File workingDirectory = new File("maven-dump-local-project");
+		String executable = "./mvnw";
+		if (SystemUtils.isWindows()) {
+			executable = Paths.get("maven-dump-local-project", "mvnw.cmd").toUri().getPath();
+		}
+		for (int i = 0; i < 3; i++) {
+			new ProcessBuilder(executable, "clean", "verify").directory(workingDirectory).start();
+		}
+		ProcessUtils.ExecutionResult result = ProcessUtils
+				.execute(new ProcessBuilder(executable, "clean", "verify").directory(workingDirectory));
+		// Get additional output for error debugging.
+		System.out.println(result.getStdout());
+		File configFile = new File(Paths.get(workingDirectory.getAbsolutePath(), "target", "tia", "agent.log").toUri());
+		String configContent = FileSystemUtils.readFile(configFile);
+		assertThat(configContent).isNotEmpty();
+		assertThat(result.terminatedByTimeoutOrInterruption()).isFalse();
+		assertThat(configContent).doesNotContain("Could not start http server on port");
```

Comment:
I was trying to let the test fail on purpose to check if it would do what it should by setting a fixed port (8087) in `TiaMojoBase#agentPort` (annotation for the default value). I noticed that the result is not deterministic. 
* Sometimes I get the expected assert error
* Sometimes I get `/Users/pete/Projects/teamscale-jacoco-agent/teamscale-jacoco-agent-2/system-tests/tia-maven/maven-dump-local-project/target/tia/agent.log (No such file or directory)` in line 38 (so the log file was not created). Seems to be a problem when cleaning the maven project, the mvn build states `Failed to clean project: Failed to delete /Users/pete/Projects/teamscale-jacoco-agent/teamscale-jacoco-agent-2/system-tests/tia-maven/maven-dump-local-project/target`. Possibly also a race condition between the clean jobs?! Might not be a problem in the build, because clean will never do anything (as we build from a fresh checkout). Maybe we can just run one clean before we do multiple builds and then just run `mvn verify` in multiple processes?
* I even managed two successful runs (which means the maven processes were not executed in parallel). Esp. this one should not happen at all. The easiest way to avoid this would be to let the tests run longer, e.g. artificially with sleeps. A clean solution would probably involve some sort of external control over the tests in the maven builds, i.e. we start all tests processes and only then tell them to finish. This is a lot more effort, ofc, and I don't really think it's worth it. WDYT @DreierF?

Reply:
Sleeps of a few seconds should typically be good enough. A low tech solution could maybe also be to let the test via busy waiting look for a file in file system and when it exists it would end. The "outer test" could create such a file, but not sure if that is necessary to make it run reliably.